### PR TITLE
feat: tally bundle proposal on skipUploaderRole

### PR DESCRIPTION
**Pull request by shifty11**
## Summary by CodeRabbit

- **New Features**
	- Introduced a function to evaluate and determine the outcome of bundle proposals based on votes.
	- Enhanced the bundle proposal submission process with comprehensive handling including tallying votes, selecting uploaders, and finalizing bundles.

- **Improvements**
	- Updated the uploader role skipping functionality to handle various bundle conditions more effectively.

- **New Types**
	- Added `TallyResultStatus` type and `TallyResult` structure to better manage bundle proposal outcomes.

### CODE REVIEW

File: x/bundles/keeper/logic_bundles.go [hunk 1]
```diff
@@ -510,3 +511,101 @@ func (k Keeper) GetVoteDistribution(ctx sdk.Context, poolId uint64) (voteDistrib
...
+	// Handle tally outcome
+	switch voteDistribution.Status {
+	case types.BUNDLE_STATUS_VALID:
+		// If a bundle is valid the following things happen:
+		// 1. Funders and Inflation Pool are charged. The total payout is divided
+		//    between the uploader, its delegators and the treasury.
+		//    The appropriate funds are deducted from the total pool funds
+		// 2. The next uploader is randomly selected based on everybody who
+		//    voted valid on this bundle.
+		// 3. The bundle is finalized by added it permanently to the state.
```
Comment (troykessler) [inline x/bundles/keeper/logic_bundles.go]: What do you mean here by `by added`?

File: x/bundles/keeper/msg_server_skip_uploader_role.go [hunk 2]
```diff
@@ -15,19 +15,43 @@ func (k msgServer) SkipUploaderRole(goCtx context.Context, msg *types.MsgSkipUpl
...
+	}
+
 	// Get next uploader, except the one who skipped
 	nextUploader := k.chooseNextUploader(ctx, msg.PoolId, msg.Staker)
 
-	bundleProposal.NextUploader = nextUploader
-	bundleProposal.UpdatedAt = uint64(ctx.BlockTime().Unix())
+	switch result.Status {
+	case types.TallyResultValid:
+		// Register the current bundle and finalize it
```
Comment (troykessler) [inline x/bundles/keeper/msg_server_skip_uploader_role.go]: we don't register anything new here, we just finalized the current proposal

**Review by troykessler — CHANGES_REQUESTED** [review]
Going through this I just realized that we can also tally a bundle proposal if the upload timeout has been reached (the next uploader did not upload within 10mins). Because right now we simply rotate the next uploader (same as in skipUploaderRole) so we can also tally the current bundle proposal here. The logic for that can be found at `x/bundles/logic_end_block_handle_upload_timeout` starting from line 70

EDIT: this should be done in a seperate PR so this comment is obsolete for this PR

File: x/bundles/keeper/logic_bundles.go (hunk 1 above)
Comment (shifty11) [reply x/bundles/keeper/logic_bundles.go]: I removed the whole comment section there because it was outdated anyway and not really necessary.

**Review by troykessler — APPROVED** [review]

**Pull request merged**